lib/shopify_toolkit/schema.rb

```diff
@@ -50,8 +52,19 @@ def load!
     end
 
     announce "Loading metafield schema from #{path}"
-    say_with_time "Executing schema statements" do
-      load path
+
+    # Parse the schema file to separate metaobject and metafield definitions
+    schema_content = File.read(path)
+
+    say_with_time "Executing metaobject definitions" do
+      # Execute only metaobject definitions first
+      execute_metaobject_definitions(schema_content)
+      apply_pending_field_validations
+    end
+
+    say_with_time "Executing metafield definitions" do
+      # Execute only metafield definitions after all metaobjects exist
+      execute_metafield_definitions(schema_content)
     end
```

Comment:
nit: you don't really "execute" a definition, would sound better as `Loading ... definitions`